### PR TITLE
[stable/redis] Rollout new pods on configuration changes

### LINKED ISSUE #2538: Redis updates secret to invalid password
**Is this a request for help?**:
No
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
Helm: 2.6.1
Kubernetes: 1.7.4

**Which chart**:
redis

**What happened**:
When deploying redis, the logic in `secrets.yaml` causes a password to be randomly generated if a password was not manually defined. When redeploying an existing redis instance, the redis chart generates a new password, which updates the value of the secret, but does not update the password redis uses to control authentication.

**What you expected to happen**:
When doing a deploy of redis, if the secret already exists, redis should use that value rather than generating a new value.

**How to reproduce it** (as minimally and precisely as possible):
1. Deploy redis
2. Check secret value
3. Authenticate to redis with that value
4. Redeploy redis
5. Check secret value
6. Attempt to authenticate to redis with new secret value (should fail)
7. Attempt to authenticate to redis with the old value (no longer in the secret) (should succeed)

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by jlegrone**
**What this PR does / why we need it**:
This PR adds annotations to redis master and slave pod templates to trigger new rollouts when the content of the configmap and secret they reference is updated.

This also fixes a bug caused when `configmap` is notated as a multiline string in the values file, but is parsed as yaml in the configmap template. Example:

```yaml
# values.yaml

## Redis config file
##
configmap: |-
  maxmemory 2mb
  maxmemory-policy allkeys-lru
```

This resulted in the following log output in all redis instances:
```cli
*** FATAL CONFIG FILE ERROR ***
Reading the configuration file, at line 1
>>> '|-'
Bad directive or wrong number of arguments
```

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:

Fixes #2538
Potentially related to #7491 

**Special notes for your reviewer**:
Note that this _will_ cause a new rollout of all redis instances for each upgrade when default values are used, and this could cause data loss if persistence is not turned on. This was a concern noted by @ctrom here: https://github.com/helm/charts/issues/2538#issuecomment-354864836

I believe it is most correct to ensure that all instances of redis deployed in this chart are always referencing the current secret contents, and this PR implements that. Dealing with `randAlphaNum` across upgrades is a [known issue](https://github.com/helm/helm/issues/3053), and at this time I don't see any way to re-use the secret contents generated during the initial chart installation.

A workaround for users of this chart who wish to prevent new rollouts for redis components during upgrades will be to provide a `password` or `existingSecret` reference as noted here: https://github.com/helm/helm/issues/3053#issuecomment-356131915

### CODE REVIEW

File: stable/redis/values.yaml [hunk 1]
```diff
@@ -203,7 +203,7 @@ master:
   ## Update strategy, can be set to RollingUpdate or onDelete by default.
   ## https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#updating-statefulsets
   statefulset:
-    updateStrategy: OnDelete
+    updateStrategy: RollingUpdate
```
Comment (jlegrone) [inline stable/redis/values.yaml:206]: Should this be considered a breaking change?

File: stable/redis/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: redis
-version: 3.8.1
+version: 3.8.2
```
Comment (Miouge1) [inline stable/redis/Chart.yaml]: Since this is a new feature i think you have to bump to 3.9.0

Comment (davidkarlsen) [reply stable/redis/Chart.yaml]: Agree 3.9.0 and I think this one is good to go

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/redis/Chart.yaml (hunk 2 above)
Comment (jlegrone) [reply stable/redis/Chart.yaml]: Works for me!

**Comment by jlegrone** [comment]
@davidkarlsen please take another look.

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**